File: objeto/metodo-construtor/cpp/Exemplo.cpp
// Disciplina: Programação Orientada a Objetos
// Professor : Gilson Pereira
// Assunto   : Métodos construtores e destrutores em C++
// Fonte     : Ascencio e Campos - Fundamentos da Programação de Computadores (3ª ed.)

#include <stdio.h>

class Produto
{
public:
    // Atributos valor e tipo
    float valor;
    char tipo;

    // Método construtor
    Produto()
    {
        printf("\nExecutando o metodo construtor");
    }

    // Método destrutor
    ~Produto()
    {
        printf("\nExecutando o metodo destrutor");
    }

    // Método que calcula o valor do imposto
    float valorImposto()
    {
        float i;

        if (valor < 500)
            i = valor * 10/100;
        else
            i = valor * 15/100;

        return i;
    }
};

int main()
{
    Produto p;
    float preco, imp;
    char tipo;

    printf("\n\nDigite o preco do produto: ");
    scanf("%f%*c", &preco);
    printf("\nDigite o tipo do produto: ");
    scanf("%c%*c", &tipo);

    p.valor = preco;
    p.tipo = tipo;

    printf("\nPreco do produto = %5.2f", p.valor);
    printf("\nTipo do produto = %c", p.tipo);

    imp = p.valorImposto();

    printf("\nImposto = %5.2f", imp);
    getchar();

    return 0;
}

File: objeto/metodo-construtor/java/Exemplo.java
// Disciplina: Programação Orientada a Objetos
// Professor : Gilson Pereira
// Assunto   : Métodos construtores e destrutores em Java
// Fonte     : Ascencio e Campos - Fundamentos da Programação de Computadores (3ª ed.)

import java.util.Scanner;

public class Exemplo
{
	//Definindo a classe Produto
	private static class Produto
	{
		public float valor;
		public String tipo;
		
		public Produto()
		{
			valor = 0;
			tipo = "";
		}
		
		public float valorImposto()
		{
			float i;
			
			if (valor < 500)
				i = valor * 10/100;
			else
				i = valor * 15/100;
			return i;
		}
	}
	
	public static void main(String args[])
	{
		Scanner dado;
		dado = new Scanner(System.in);
		
		Produto p = new Produto();
		
		float preco, imp;
		String tipo;
		
		System.out.println("digite o preço do produto: ");
		preco = dado.nextFloat();
		
		System.out.println("digite o tipo do produto: ");
		tipo = dado.next();
		
		p.valor = preco;
		p.tipo = tipo;
		
		System.out.println("\nPreço do produto = " + p.valor);
		System.out.println("\nTipo do produto = " + p.tipo);
		
		imp = p.valorImposto();
		System.out.println("\nImposto = " + imp);
	}
}